### PR TITLE
Add search API

### PR DESCRIPTION
**Pull request by giang-nghg**

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        IdentityFile /root/.ssh/id_github
+        EOF
+        ssh-keyscan bitbucket.org >> /root/.ssh/known_hosts
+        ssh-keyscan github.com >> /root/.ssh/known_hosts
+    volumes:
+      - name: "ssh"
+        path: /root/.ssh
+
+  # Build and push API image
+  - name: "gcr.io/cloud-builders/docker"
```
Comment (Zhicheng-Liu) [inline cloudbuild.yaml:50]: Could you please add test before building the Docker image?

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

File: src/openapi_server/controllers/search_controller.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    :param search_query: 
+    :type search_query: dict | bytes
+
+    :rtype: SearchResults
+    """
+    if connexion.request.is_json:
+        search_query = SearchQuery.from_dict(connexion.request.get_json())  # noqa: E501
+
+    results = cobs.search(search_query.seq, search_query.threshold)
+    num_kmers = ceil(len(search_query.seq) / COBS_TERM_SIZE)
```
Comment (Zhicheng-Liu) [inline src/openapi_server/controllers/search_controller.py]: It should be `num_kmers = len(search_query.seq) - COBS_TERM_SIZE + 1`.

File: src/openapi_server/controllers/search_controller.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    :param search_query: 
+    :type search_query: dict | bytes
+
+    :rtype: SearchResults
+    """
+    if connexion.request.is_json:
+        search_query = SearchQuery.from_dict(connexion.request.get_json())  # noqa: E501
+
+    results = cobs.search(search_query.seq, search_query.threshold)
```
Comment (Zhicheng-Liu) [inline src/openapi_server/controllers/search_controller.py:25]: Please make sure `len(search_query.seq) >= COBS_TERM_SIZE`.

File: specs/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+paths:
+  /search:
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SearchQuery'
+        required: true
+      responses:
```
Comment (Zhicheng-Liu) [inline specs/openapi.yaml:16]: Sorry, I think we should also return 400 if the seq length is less than the term size etc.

**Review by Zhicheng-Liu — CHANGES_REQUESTED** [review]

File: src/openapi_server/controllers/search_controller.py (hunk 2 above)
Comment (giang-nghg) [reply src/openapi_server/controllers/search_controller.py]: I don't understand, isn't that number of base letters, not number of k-mers (k == term_size)?

Comment (Zhicheng-Liu) [reply src/openapi_server/controllers/search_controller.py]: I can refer you to this page for how kmers are derived from a sequence: https://en.wikipedia.org/wiki/K-mer

**Pull request merged**